dask_cuda/tests/test_dask_cuda_worker.py

```diff
@@ -42,7 +45,49 @@ def get_visible_devices():
 
                     # verify 4 workers with the 4 expected CUDA_VISIBLE_DEVICES
                     result = client.run(get_visible_devices)
-                    expected = {"2,3,7,8": 1, "3,7,8,2": 1, "7,8,2,3": 1, "8,2,3,7": 1}
+                    expected = {"0": 1}
+                    for v in result.values():
+                        del expected[v]
+                    assert len(expected) == 0
+    finally:
+        del os.environ["CUDA_VISIBLE_DEVICES"]
```

Comment:
I think that it would be nice to maintain the previous test.  It sounds like you're running into issues because now we need to have GPUs in order to not err with LocalCUDACluster.  As an alternative, I recommend that you keep the old tests, but set `worker_class=Nanny` so that we don't pick up the CUDAWorker code that requires the presense of a GPU.

Sorry to mention this late.  I should have inspected this code more deeply earlier.

Reply:
I can keep both, if you'd prefer, but I think the old test is really pointless for the following reasons:

1. CI runs on a single GPU at the moment, so testing would happily pass even if you had any random index in `CUDA_VISIBLE_DEVICES`, so that means if you had `CUDA_VISIBLE_DEVICES=1000000`, it would still pass;
1. The new `worker_class=CUDANanny` requires proper index of GPUs in `CUDA_VISIBLE_DEVICES` to be able to query how much memory there is available.

Reply:
> CI runs on a single GPU at the moment, so testing would happily pass even if you had any random index in CUDA_VISIBLE_DEVICES, so that means if you had CUDA_VISIBLE_DEVICES=1000000, it would still pass;

Right, but we were testing that the distribution of GPUs was nicely cyclic.  We cared that every process had access to each GPU, but that it preferred one in particular.

```python
expected = {"2,3,7,8": 1, "3,7,8,2": 1, "7,8,2,3": 1, "8,2,3,7": 1}	
```

> The new worker_class=CUDANanny requires proper indexing of GPUs in CUDA_VISIBLE_DEVICES to be able to query how much memory there is available.
 
Understood.  This sentence was intended to address this.

> I recommend that you keep the old tests, but set `worker_class=Nanny` so that we don't pick up the `CUDAWorker` code that requires the presense of a GPU.

Reply:
Just FYI. I'm looking into adding the test back, but I realized we'll need some changes to the `dask-cuda-worker` executable, i.e., a `--worker-class` argument and a few more changes that come with it.

Reply:
I'm not sure there is a good way to expose the worker class parameter through the CLI today.  What were you planning?

Reply:
There was no way, but I exposed it via CLI argument. Take a look at the new changes.